frontend/src/metabase/admin/people/components/GroupsListing.jsx

```diff
@@ -56,32 +60,67 @@ function AddGroupRow({ text, onCancelClicked, onCreateClicked, onTextChange }) {
 
 // ------------------------------------------------------------ Groups Table: editing ------------------------------------------------------------
 
-function DeleteGroupModal({ group, onConfirm = () => {}, onClose = () => {} }) {
+function DeleteGroupModal({
+  group,
+  apiKeys,
+  onConfirm = () => {},
+  onClose = () => {},
+}) {
+  const apiKeysCount = apiKeys.length;
+  const hasApiKeys = apiKeys.length > 0;
+
+  const modalTitle =
+    apiKeysCount === 0
+      ? t`Remove this group?`
+      : apiKeysCount === 1
+      ? t`Are you sure you want remove this group and its API key?`
+      : t`Are you sure you want remove this group and its API keys?`;
+
+  const confirmButtonText =
+    apiKeysCount === 0
+      ? t`Remove group`
+      : apiKeysCount === 1
+      ? t`Remove group and API key`
+      : t`Remove group and API keys`;
+
   return (
-    <ModalContent title={t`Remove this group?`} onClose={onClose}>
-      <p className="px4 pb4">
-        {t`Are you sure? All members of this group will lose any permissions settings they have based on this group.
+    <ModalContent title={modalTitle} onClose={onClose}>
+      <Stack spacing="xl">
+        <Text>
+          {hasApiKeys
+            ? jt`All members of this group will lose any permissions settings they have based on this group, and its related API keys will be deleted. You can ${(
```

Comment:
I found that when I associated an API key with a group, and then deleted that group, the API key became associated with 'All Users' - is that expected?

Here's a [screencast](https://www.loom.com/share/15b81b58f94a4868957a4b9e3355ee27?sid=44d9feda-a482-4e82-98e3-fd3156f0036e) illustrating this (note, it has no audio)

Reply:
this is an interesting point - by definition all users are part of the all users group - but some of the UI doesn't reflect that very well. the user count endpoint only returns api keys that are *only* part of the all users group, but the user list shows all api keys. I'll follow up with design on the intended behavior, but I think this will be a backend issue.